include/__generator.hpp

```diff
@@ -451,18 +461,20 @@ struct __generator_promise_base
         return std::move(__g).get();
     }
 
-    template <std::ranges::range _Rng>
-    __yield_sequence_awaiter<generator<_Ref>> yield_value(std::ranges::elements_of<_Rng> __x) {
+    template <std::ranges::range _Rng, typename _Allocator>
+    __yield_sequence_awaiter<generator<_Ref, std::ranges::range_value_t<_Rng>, _Allocator>> yield_value(std::ranges::elements_of<_Rng, _Allocator> __x) {
         // TODO: Consider propagating parent coroutine's allocator to child generator
         // coroutine here.
-        return yield_value(std::ranges::elements_of([](_Rng&& __rng) -> generator<_Ref> {
+
+        // TODO: Should the promise type be templated on value to reduce template instantiations?
```

Comment:
The promise type kind of needs to be templated on value so that `get_return_object()` can return a `generator` of the correct type.

However, it may be possible to have `get_return_object()` return some proxy object that is implicitly convertible to `generator<Ref, *, Alloc>` so that we don't have to have the promise-type depend on value_type.

Reply:
Correct me if i,'m wrong but there is no way for a user to access a yielded nested generator, so, whatever type we 'd use as the value type would be "correct", right?

I wonder if 
```cpp
template <typename _Ref, typename_Alloc>
class __generator {
};

template <typename _Ref, typename _Value, typename _Alloc>
class generator : public __generator<_Ref, _Alloc> {
    struct iterator {
          using value_type = _Value;
    };
};
```
would be possible / worth exploring

Reply:
Yes, that could be a good code-size win.
Created #10 to track this.